### PR TITLE
Dialogfragment

### PR DESCRIPTION
**Pull request by paulpv**
Proposed fix for #14 (and a few other updates)

### CODE REVIEW

**Comment by paulpv** [comment]
Can you update your travis-ci to have build tools 23.0.3?

**Comment by greysonp** [comment]
If you update the [.travis.yml](https://github.com/greysonp/permiso/blob/master/.travis.yml) in this PR, will Travis not pick it up? If not, I'll make the change later tonight.
(Thanks for the PR, btw!)

**Comment by paulpv** [comment]
Didn't seem to make any diff; I'm a long time Jenkins user, but don't know much about the areas of Travis that are different from Jenkins.

**Comment by greysonp** [comment]
Yeah no that makes sense, it probably always uses the config on master.  I've updated it.

**Comment by greysonp** [comment]
So I got master building on version 23.0.3. To get it to work I had to add a couple things. Make it match [master](https://github.com/greysonp/permiso/blob/master/.travis.yml) and you should be fine.

File: permiso/src/main/java/com/greysonparrelli/permiso/Permiso.java [hunk 1]
```diff
@@ -178,9 +180,17 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
...
     public void showRationaleInDialog(@Nullable String title, @NonNull String message, @Nullable String buttonText, @NonNull final IOnRationaleProvided rationaleCallback) {
-        checkActivity();
+        Activity activity = checkActivity();
 
-        PermisoDialogFragment dialogFragment = PermisoDialogFragment.newInstance(title, message, buttonText);
+        FragmentManager fm = activity.getFragmentManager();
+
+        PermisoDialogFragment dialogFragment = (PermisoDialogFragment) fm.findFragmentByTag(PermisoDialogFragment.TAG);
+        if (dialogFragment != null)
+        {
```
Comment (greysonp) [inline permiso/src/main/java/com/greysonparrelli/permiso/Permiso.java]: Same-line curly brace please :)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -6,5 +6,5 @@
     <string name="permission_granted">Permission Granted!</string>
     <string name="permission_permanently_denied">Permission Permanently Denied.</string>
     <string name="permission_denied">Permission Denied.</string>
-    <string name="two_permission_granted">2 Permissions Granted.</string>
+    <string name="X_permissions_granted">%1$d Permissions Granted.</string>
```
Comment (greysonp) [inline app/src/main/res/values/strings.xml]: Please use lowercase 'x'.

**Comment by greysonp** [comment]
I'd like to apologize for not seeing your update to this PR. After I suggested to merge in master, I somehow never got notified of your new commit, and this all got lost. Added a couple comments, just style things. If you don't see this or aren't motivated after I caused such a long delay, I'll just merge it and make the tweaks myself :)

**Pull request merged**